tsdb/compact.go

```diff
@@ -191,14 +195,15 @@ func NewLeveledCompactorWithOptions(ctx context.Context, r prometheus.Registerer
 		pe = index.EncodePostingsRaw
 	}
 	return &LeveledCompactor{
-		ranges:                   ranges,
-		chunkPool:                pool,
-		logger:                   l,
-		metrics:                  newCompactorMetrics(r),
-		ctx:                      ctx,
-		maxBlockChunkSegmentSize: maxBlockChunkSegmentSize,
-		mergeFunc:                mergeFunc,
-		postingsEncoder:          pe,
+		ranges:                       ranges,
+		chunkPool:                    pool,
+		logger:                       l,
+		metrics:                      newCompactorMetrics(r),
+		ctx:                          ctx,
+		maxBlockChunkSegmentSize:     maxBlockChunkSegmentSize,
+		mergeFunc:                    mergeFunc,
+		postingsEncoder:              pe,
+		disableOverlappingCompaction: opts.DisableOverlappingCompaction,
```

Comment:
We prefer positive flags, so this should be `enableOverlappingCompaction`.

Reply:
:bow: thanks for the review, I've updated it to be `enableOverlappingCompaction`.